### DISCUSSION TITLE
如何禁止游客下载

### DISCUSSION THREAD

**Discussion by JulienZeng** [AlistGo/alist#1097] [General]
如题，有一个文件夹，内容如何只让游客上传而不让游客下载呢

**Comment by anwen-anyi** [comment]
自己改源码..QAQ..如果不会就没辙了 哈哈

**Comment by zyk121381** [comment]
如果不希望用户下载，同时文档图片什么的也不对用户开放（只开放文件列表），那可以再代理URL随意填写。如果希望文档图片什么的游客可以正常预览，仅仅只是限制下载功能的话，可以通过在alist后台全局设置的自定义头部自己写css文件，把下拉下载的按钮隐藏。如果本身没有预览窗格，点开文件即下载，那么可以写css将这两个按钮隐藏。

提示：用户长按（移动端）/右键（PC）文件列表也可调出复制链接和下载按钮。如果需要防止用户下载，记得把这两个也写到css里面

**Comment by Wallemuse** [comment]
能看就能下载，总有办法下载的吧。比如装浏览器插件或油猴插件

**Reply by zyk121381** [reply]
照这样说的话，世界上就没有什么方法可以允许游客查看但禁止下载了。因为甚至不需要插件或脚本，直接在浏览器控制台打开就能很方便地获得文件的下载链接。
